server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -418,24 +434,16 @@ public boolean equals(Object obj) {
                 return false;
             }
             TimeUnitRounding other = (TimeUnitRounding) obj;
-            return Objects.equals(unit, other.unit) && Objects.equals(timeZone, other.timeZone);
+            return unit == other.unit && timeZone.equals(other.timeZone);
```

Comment:
I'll revert this change. It snuck in because I thought the comparison was wrong.